### ISSUE TITLE
Page title not using "getTitle" from custom theme.

### ISSUE THREAD

**Issue by fancycode** [nextcloud/server#5374]
### Steps to reproduce
1. Use a custom theme that implements `getTitle` to return a custom string. `getName` still returns `Nextcloud`.
2. Login to Nextcloud and open any page (e.g. the admin settings).

### Expected behaviour
The page title in the browser should be `Administrator - <mycustomtitle>`.

### Actual behaviour
The page title in the browser is `Administrator - Nextcloud`.

### Server configuration

**Nextcloud version:** (see Nextcloud admin page)
12.0.0

**Updated from an older Nextcloud/ownCloud or fresh install:**
Upgraded from 10.0.1 (via 11.0.3).

The page title was correct with Nextcloud 10 and 11.

**Comment by FlorianTaut** [comment]
Do you mean the Textbox in the Theming app or self created theme on the server directly?
For me it works if i change it in the settings

**Comment by fancycode** [comment]
I mean a custom theme in the `themes` subfolder on the server (methods implemented in `defaults.php` in the theme).

**Comment by juliushaertl** [comment]
@fancycode I cannot reproduce this with the example theme shipped with Nextcloud 12. Is the theming app disabled?

**Comment by fancycode** [comment]
The theming app is enabled, for now I could "fix" my problem with this patch:
```patch
--- a/apps/theming/lib/ThemingDefaults.php
+++ b/apps/theming/lib/ThemingDefaults.php
@@ -44,6 +44,8 @@
 	/** @var string */
 	private $name;
 	/** @var string */
+	private $title;
+	/** @var string */
 	private $url;
 	/** @var string */
 	private $slogan;
@@ -79,6 +81,7 @@
 		$this->util = $util;
 
 		$this->name = parent::getName();
+		$this->title = parent::getTitle();
 		$this->url = parent::getBaseUrl();
 		$this->slogan = parent::getSlogan();
 		$this->color = parent::getColorPrimary();
@@ -93,7 +96,7 @@
 	}
 
 	public function getTitle() {
-		return $this->getName();
+		return strip_tags($this->config->getAppValue('theming', 'name', $this->title));
 	}
 
 	public function getEntity() {
```

We are packaging Nextcloud for the @spreedbox and ship it with our custom theme included and activated in `config.php`. This has worked fine with the previous versions as mentioned above and now breaks with Nextcloud 12.

**Comment by juliushaertl** [comment]
@fancycode Ok, so you are using a setTitle method but no getName method? That patch looks good, do you want to open a PR for that?

**Comment by fancycode** [comment]
Sure, I can open a PR, I just was not sure if this is the intended behaviour ;-) A similar issue is with the `getEntity` method which can be implemented by custom themes, but will also not be used as the `ThemingDefaults` will delegate to `getName` in their `getEntity` method.

**Comment by juliushaertl** [comment]
Having the theming app enabled with a custom theme has always been not recommended, but I think we should move to make them work together. Therefore this is a good change. :wink:

**Comment by fancycode** [comment]
Maybe it would also be good change then to not enable the theming app by default if a custom theme is configured during initial setup :wink:

Anyway, this change solves the problem I had, so I'm happy to use / upstream it.

**Comment by MorrisJobke** [comment]
Fixed with #5398
